status-handler.js

```diff
@@ -143,6 +147,70 @@ class StatusHandler {
     this.log.info(`Ignored status event, no build occurred`, data);
   }
 
+  /**
+   * Check if there are third-party apps that
+   * subscribed for webhooks for a specific package
+   *
+   * @function _shouldSendWebhook
+   * @param {String} pkg - The name of the package
+   * @returns {Boolean} the computed value
+   */
+  _shouldSendWebhook(pkg) {
+    const webhooks = this.webhooks[pkg];
+    if (!webhooks || webhooks.length === 0) {
+      this.log.info(`No webhooks for pkg ${pkg}`);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Process and dispatch 'build_started' webhook
+   *
+   * @function _sendBuildStartedWebhook
+   * @param {Object} data - Message from NSQ
+   * @returns {Promise} to resolve
+   */
+  async _sendBuildStartedWebhook(data) {
+    const { pkg, name, version, env } = data;
+    const pkgName = pkg || name;
+
+    if (!this._shouldSendWebhook(pkgName)) return;
+
+    // Fetch list of previous events for a certain build
+    const previousEvents = await this.models.Status.findAll({
+      pkg: pkgName,
+      version,
+      env
+    });
+
+    // Ensure that last event is the 'queued' event from carpentd
+    const lastEvent = previousEvents[previousEvents.length - 1];
+    if (!lastEvent || lastEvent.message === 'Builds Queued') return;
+
+    // Prepare webhook body payload and send it to subscribed third-parties
+    const body = { event: 'build_started', pkg: pkgName, version, env };
+    return this._sendWebhook(body);
+  }
+
+  /**
+   * Send webhook to third-party applications
+   *
+   * @function _sendWebhook
+   * @param {Object} body - Webhook body payload
+   * @returns {Promise} to resolve
+   */
+  async _sendWebhook(body) {
+    const webhooks = this.webhooks[body.pkg];
+    try {
+      const params = { body, method: 'POST', json: true };
+      await Promise.all(webhooks.map(uri => request({ uri, ...params })));
+    } catch (err) {
+      this.log.error('Status Handler errored %s', err.message, body);
+    }
+    return Promise.resolve(); // hack valid-jsdoc
```

Comment:
jsdoc should be smarter than this :(

Reply:
Nah it's not. See issue https://github.com/eslint/eslint/issues/7307.